File: clothes/c/sol.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

float min(float x, float y) {
    if(x > y) return y;
        else return x;
}

int main() {
    int nb_items = 3, nb_pairs = 3;
    float prices[] = {1, 2, 3};
    float matching_pairs[3][2] = {{1,2}, {2, 3}, {3, 1}};

    if (nb_items < 3 || nb_pairs < 3) {
        printf("-1\n");
        exit(0);
    }

    float matching_costs[3][3] = {INFINITY}; 

    for (int i =0; i < nb_pairs; i++) {
        int i = matching_pairs[i][0] - 1, j = matching_pairs[i][1] - 1;
        matching_costs[i][j] = min(prices[i], prices[j]);
        matching_costs[j][i] = min(prices[i], prices[j]);
    }

    float min_cost = INFINITY;

    for (int i = 0; i < nb_items; i++) {
        for (int j = i + 1; j < nb_items; j++) {
            if (matching_costs[i][j] == INFINITY) continue;
            for (int k = j + 1; k < nb_items; k++) {
                if (matching_costs[i][j] != INFINITY && matching_costs[j][k] != INFINITY) {
                    min_cost = min(min_cost, prices[i] + prices[j] + prices[k]);
                }
            }
        }
    }
    if (min_cost == INFINITY) {
        printf("-1\n");
    } else {
        printf("%.15g\n", min_cost);
    }
    return 0;
}

File: clothes/sol.java
package clothes;

import java.text.DecimalFormat;
import java.util.Arrays;
import java.util.Scanner;

/**
 * sol
 */
public class sol {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int[] first_line = Arrays.stream(scanner.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray(); // number
                                                                                                               // of
                                                                                                               // items
                                                                                                               // &
                                                                                                               // number
                                                                                                               // of
                                                                                                               // matching
                                                                                                               // pairs
        double[] prices = Arrays.stream(scanner.nextLine().split(" ")).mapToDouble(Double::parseDouble).toArray();
        int[][] matchingPairs = new int[first_line[1]][2];

        for (int i = 0; i < first_line[1]; i++)
            matchingPairs[i] = Arrays.stream(scanner.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();

        double result = solution(first_line[0], first_line[1], prices, matchingPairs);
        System.out.println(new DecimalFormat("#").format(result));

        scanner.close();
    }

    public static double solution(int nbItems, int nbMatchingPairs, double[] prices, int[][] matchingPairs) {
        if (nbItems < 3 || nbMatchingPairs < 3)
            return -1;

        // init
        double[][] matchingCosts = new double[nbItems][nbItems];
        for (double[] array : matchingCosts)
            Arrays.fill(array, Double.POSITIVE_INFINITY);

        for (int[] matchingPair : matchingPairs) {
            int i = matchingPair[0] - 1;
            int j = matchingPair[1] - 1;
            matchingCosts[i][j] = Math.min(prices[i], prices[j]);
            matchingCosts[j][i] = Math.min(prices[i], prices[j]);
        }

        double min_cost = Double.POSITIVE_INFINITY;
        for (int i = 0; i < nbItems; i++)
            for (int j = i + 1; j < nbItems; j++) {
                if (matchingCosts[i][j] == Double.POSITIVE_INFINITY)
                    continue;
                for (int k = j + 1; k < nbItems; k++)
                    min_cost = Math.min(min_cost, prices[i] + prices[j] + prices[k]);
            }

        if (min_cost == Double.POSITIVE_INFINITY)
            return -1;
        else
            return min_cost;
    }

}